test/with_dummyserver/test_socketlevel.py

```diff
@@ -1643,6 +1643,120 @@ def socket_handler(listener: socket.socket) -> None:
             ]
             assert expected_response_headers == actual_response_headers
 
+    def _simple_read_http_request(self, sock: socket.socket) -> Tuple[(bytes, bytes)]:
```

Comment:
This is kind of messy, but honestly I find this a more satisfying solution than sprinkling timeouts into the socket handlers to try and hope that the body doesn't get in the way.